independent-projects/bootstrap/maven-resolver/src/test/java/io/quarkus/bootstrap/resolver/maven/test/ChainedLocalRepositoryManagerTest.java

```diff
@@ -0,0 +1,206 @@
+                Paths.get(M2_LOCAL_1, "org", "acme", "foo", "0.0.1", "foo-0.0.1.jar").toAbsolutePath().toString());
+
+        final Artifact bar = resolveOrgAcmeBarJar002(mvn);
+        assertNotNull(bar);
+        assertEquals(bar.getFile().getAbsolutePath(),
+                Paths.get(M2_LOCAL_2, "org", "acme", "bar", "0.0.2", "bar-0.0.2.jar").toAbsolutePath().toString());
+    }
+
+    // ignoreAvailability tests
+
+    @Test
+    public void testValidTailLocalCheckingForAvailabilityViaConfig() throws Exception {
+        final BootstrapMavenContext mvn = bootstrapMavenContextForProject("workspace-with-local-repo-tail",
+                BootstrapMavenContext.config()
+                        .setLocalRepositoryTailIgnoreAvailability(false)
+                        .setLocalRepositoryTail(new String[] { M2_LOCAL_1 }));
+
+        assertNotNull(resolveOrgAcmeFooJar001(mvn));
+    }
+
+    @Test
+    public void testValidTailFromRemoteCheckingForAvailabilityViaConfig() throws Exception {
+        final BootstrapMavenContext mvn = bootstrapMavenContextForProject("workspace-with-local-repo-tail",
+                BootstrapMavenContext.config()
+                        .setLocalRepositoryTailIgnoreAvailability(false)
+                        .setLocalRepositoryTail(new String[] { M2_FROM_REMOTE }));
+
+        assertThrowsExactly(BootstrapMavenException.class, () -> resolveOrgAcmeFooJar001(mvn));
+    }
+
+    @Test
+    public void testValidTailFromRemoteCheckingForAvailabilityViaSystemProp() throws Exception {
+        setSystemProp("maven.repo.local.tail.ignoreAvailability", "false");
+        final BootstrapMavenContext mvn = bootstrapMavenContextForProject("workspace-with-local-repo-tail",
+                BootstrapMavenContext.config()
+                        .setLocalRepositoryTail(new String[] { M2_FROM_REMOTE }));
+
+        assertThrowsExactly(BootstrapMavenException.class, () -> resolveOrgAcmeFooJar001(mvn));
+    }
+
+    @Test
+    public void testValidTailFromRemoteIgnoringAvailabilityViaSystemPropEmpty() throws Exception {
+        setSystemProp("maven.repo.local.tail.ignoreAvailability", ""); // will become `true`
+        final BootstrapMavenContext mvn = bootstrapMavenContextForProject("workspace-with-local-repo-tail",
+                BootstrapMavenContext.config()
+                        .setLocalRepositoryTail(new String[] { M2_FROM_REMOTE }));
+
+        assertNotNull(resolveOrgAcmeFooJar001(mvn));
+    }
+
+    @Test
+    public void testValidTailFromRemoteIgnoringAvailabilityViaSystemPropBlank() throws Exception {
+        setSystemProp("maven.repo.local.tail.ignoreAvailability", " "); // will become `true`
+        final BootstrapMavenContext mvn = bootstrapMavenContextForProject("workspace-with-local-repo-tail",
+                BootstrapMavenContext.config()
+                        .setLocalRepositoryTail(new String[] { M2_FROM_REMOTE }));
+
+        assertNotNull(resolveOrgAcmeFooJar001(mvn));
```

Comment:
@tiagobento could you please explain this one. `ignoreAvailability` is `true`. Is it availability of artifacts that should be ignored?

Reply:
Sure. So maybe the `ignoreAvailability` name kind of gives the wrong idea of what it actually does, and maybe a more explicit, albeit verbose, name would be `ignoreRemoteAvailability`? cc @cstamas

This flag is only applicable for dependencies that have been downloaded from a remote repository prior to being consumed as part of a `tail` local repository. See the `_remote.repositories` file.

This particular test succeeds because `maven.repo.local.tail.ignoreAvailability` ends up being `true`, so even though org.acme:foo:jar:0.0.1 was supposedly initially resolved from a remote repository with id = `some-repo-id`, this fact is ignored, and it being available in the local repository suffices. When `maven.repo.local.tail.ignoreAvailability` is false, it fails. That case is exactly what is covered by `testValidTailFromRemoteCheckingForAvailabilityViaSystemProp`.

Reply:
Thanks for the explanation. The property name is indeed very confusing. It sounds like the meaning is "trust any remote repo" that provided an artifact.

Reply:
"availability" in resolver means that tracked remote repository (by repoID) is available in the context artifact is asked for. So "remote" may come in as "remote repo ID is available in current context asking for artifact" :)